### PR TITLE
Near finalized commit for backend searching

### PR DESCRIPTION
**Pull request by cphalen**
Mainly re-purposed the `resetDisplay` method in `index.tsx`. Lots of kinda convoluted logic for parsing the checkboxes and turning it into the correct backend query -- let me know if you have ideas on how to clean this up.

Also, made a few small changes to `views.py`. Namely, fixing a bug on parsing the operation, as well as adding supporting for parsing multiple ints (although I think you also added in a solution for this concurrently, when merging I just kept both for good measure because I didn't want to break anything else you might've been working on).

### CODE REVIEW

File: backend/clubs/views.py [hunk 1]
```diff
@@ -205,13 +205,19 @@ def parse_year(field, value, operation, queryset):
             return {}
 
         def parse_int(field, value, operation, queryset):
+            if operation == "in":
+                values = value.strip().split(",")
+                sizes = [int(size) for size in values if size]
+                return {f"{field}__in": sizes}
+
```
Comment (cphalen) [inline backend/clubs/views.py:212]: My solution to the same problem that you solve below, instead I added a "in" operation. We can go with either solution if you wanna take mine out no worries.

File: frontend/pages/index.tsx [hunk 2]
```diff
@@ -159,43 +159,61 @@ class Splash extends React.Component<SplashProps, SplashState> {
...
+        (tag) => tag.name === 'Application' && tag.value === 1,
+      ) !== -1
+    const noRequiredApplication =
+      selectedTags.findIndex(
+        (tag) => tag.name === 'Application' && tag.value === 2,
+      ) !== -1
+    const acceptingMembers =
+      selectedTags.findIndex(
+        (tag) => tag.name === 'Application' && tag.value === 3,
+      ) !== -1
```
Comment (cphalen) [inline frontend/pages/index.tsx:180]: Kinda messy but I couldn't come up with a better way to do this

File: backend/clubs/views.py (hunk 1 above)
Comment (ezwang) [reply backend/clubs/views.py:212]: The default operation is actually "in" if you don't specify "and", see line 214/219 below.

I like how you made it explicit though! Definitely makes it less confusing for someone reading the code. One option could be to add a comment above line 214/219 saying that the default behavior is to do "in".

File: frontend/pages/index.tsx (hunk 2 above)
Comment (ezwang) [reply frontend/pages/index.tsx:180]: Hmm, I can't think of a better way to do this either without refactoring more broadly. The whole sidebar tag system could be refactored a bit to make it easier to extend, but that can be done in a later commit.

This works for me!

**Pull request merged**